lib/app/analytics.coffee

```diff
@@ -3,8 +3,14 @@ Analytics = require 'analytics-node'
 
 @setCampaign = (req, res, next) ->
   return next() unless opts.SEGMENT_WRITE_KEY
-  req.session.modalId = req.body.modal_id
-  req.session.acquisitionInitiative = req.body.acquisition_initiative
+  req.session.modalId = (
+    req.body.modal_id or
+    req.query.modal_id
+  )
+  req.session.acquisitionInitiative = (
```

Comment:
We want to be able to track the type of marketing modal a user lands on, so we allow sending these `acquisition_initiative: 'miami` params along with the signup data like `email: 'foo@bar.com'`. We have to store this in the session b/c OAuth is this crazy protocol that involves redirecting all over various domains. To be able to persist this state when the user lands back in Artsy we use the session to store it and then later in `@trackSignup` we send that along with other data in the server-side Segment call.

What I forgot is that we _can't_ send this through the request body b/c OAuth (like Facebook Signup) is just a link that does a bunch of redirecting—so we need to pass that through query params. More [reading here](https://github.com/artsy/artsy-passport/pull/86#discussion_r89008080).